packages/shared/src/dto/workflows/workflow-commons-fields.ts

```diff
@@ -1,13 +1,35 @@
 import { IsArray, IsBoolean, IsDefined, IsOptional, IsString } from 'class-validator';
+import { JSONSchema } from 'json-schema-to-ts';
 import { WorkflowResponseDto } from './workflow-response-dto';
 import { Slug, StepTypeEnum, WorkflowPreferences } from '../../types';
+import { StepContentIssueEnum, StepIssueEnum } from './step-content-issue.enum';
 
+export class ControlsSchema {
+  schema: JSONSchema;
+}
+export type StepCreateAndUpdateKeys = keyof StepCreateDto | keyof StepUpdateDto;
```

Comment:
it's not a tech debt at all, it was super international in order to avoid wrongly optional values 
![image](https://github.com/user-attachments/assets/584f9658-08c4-4a92-ac8f-39fa3def388d)
we have everything you described, super intentional, can be discussed, but I don't find any debt here